docs/reference/hooks.md

```diff
@@ -0,0 +1,154 @@
+---
+layout: default
+title: Hooks
+parent: Reference
+nav_order: 10
+has_children: false
+---
+
+# Configurable Hooks
+Like other version control systems, lakeFS allows the configuration of `Actions` to trigger when predefined events occur.
+ 
+Supported Events:
+1. `pre_commit` - Action runs when the commit occurs, before the commit is finalized.
+2. `pre_merge` - Action runs when the merge occurs, before the merge is finalized.
+ 
+lakeFS `Actions` are handled per repository and cannot be reused.  
+Failure of any `Hook` under any `Action` will result in aborting the lakeFS operation that is taking place.
+
+## Action
+An `Action` is a list of `Hooks` with the same trigger configuration.
+`Hooks` list under an `Action` is ordered and an `Hook` will only be executed if all previous `Hooks` passed.  
+
+### Hook
+An `Hook` is the basic building block of an `Action`. 
+Failure of a single `Hook` will stop the execution of the containing `Action` and fail the `Run`. 
+
+### Action file
+Schema of the Action file:
+
+|Property          |Description                                            |Required |Default Value
+|------------------|-------------------------------------------------------|---------|--------------------------------------|
+|name              |Identify the Action file                               |false    | If missing, filename is used instead 
+|on                |List of events that will trigger the hooks             |true     |
```

Comment:
please specify data type (string? array of strings? enum of certain values?)

Reply:
Done.